app/models/krikri/provider.rb

```diff
@@ -157,7 +157,8 @@ def initialize_name
                        :fl => 'provider_name' }
       response = Krikri::SolrResponseBuilder.new(query_params).response
       return nil unless response.docs.count > 0
-      response.docs.first['provider_name'].first
+      response.docs.first['provider_name'].respond_to?(:first) ?
+        response.docs.first['provider_name'].first : nil
```

Comment:
Good idea!  This way, the `id` will get displayed in the nav if the provider's name is `nil`.